x/group/types/conditions.go

```diff
@@ -0,0 +1,132 @@
+package types
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/regen-network/regen-ledger/orm"
+)
+
+var (
+	// AddressLength is the length of all addresses
+	// You can modify it in init() before any addresses are calculated,
+	// but it must not change during the lifetime of the kvstore
+	AddressLength = 20
+
+	// it must have (?s) flags, otherwise it errors when last section contains 0x20 (newline)
+	perm = regexp.MustCompile(`(?s)^([a-zA-Z0-9_\-]{3,8})/([a-zA-Z0-9_\-]{3,8})/(.+)$`)
+)
+
+// Condition is a specially formatted array, containing
+// information on who can authorize an action.
+// It is of the format:
+//
+//   sprintf("%s/%s/%s", extension, type, data)
+type Condition []byte
```

Comment:
Why don't we we represent a condition using a structure with 3 fields: extension, type, data? We can have 2 methods: `String()` and `MarshalJSON`, and one additional constructor `ParseCondition(str)`

Reply:
TBH I haven't really touched this part of the code that was pre-existing in cosmos-modules. A group account address is directly derived from a `Condition` so I guess that's why it was originally designed as `[]byte`. I agree using a struct would be cleaner. That being said, that part will need to be revisited as part of ADR 028 and ADR 033 eventually...